### PR TITLE
luci-proto-openfortivpn: add luci interface for openfortivpn

### PR DESCRIPTION
**Pull request by aaronjg**
Create a luci interface for the openfortivpn netifd protocol.

@jow- Is there anyway to exclude the current interface name in the `widgets.NetworkSelect` selector? I'd like to make it impossible for the user to accidentally try to use their new vpn interface as the bound interface.

cc: @lucize 
Signed-off-by: Aaron Goodman <aaronjg@stanford.edu>

### CODE REVIEW

**Comment by jow-** [comment]
> Is there anyway to exclude the current interface name in the widgets.NetworkSelect selector?

Try `o.exclude = s.section;`

File: protocols/luci-proto-openfortivpn/htdocs/luci-static/resources/protocol/openfortivpn.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+		o = s.taboption('general', form.Value, 'password', _('Password'));
+		o.password = true;
+
+		o = s.taboption('advanced', widgets.NetworkSelect, 'iface_name', _('Bind interface'), _('Bind the tunnel to this interface (optional).'));
+		o.nocreate = true;
+		o.optional = true;
+
+		s.taboption('advanced', form.Value, 'trusted_cert', _("VPN Server's certificate SHA1 hash"));
+		o.nocreate = true;
```
Comment (jow-) [inline protocols/luci-proto-openfortivpn/htdocs/luci-static/resources/protocol/openfortivpn.js:63]: Remove `o.nocreate` from this and the subsequent options

File: protocols/luci-proto-openfortivpn/htdocs/luci-static/resources/protocol/openfortivpn.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+		o.optional = true;
+
+
+		o = s.taboption('advanced', form.Flag, 'set_dns', _('Set DNS'));
+		o.default = o.enabled;
+		o.nocreate = true;
+		o.optional = true;
+
+		o = s.taboption('advanced', form.Flag, 'pppd_use_peerdns', _('PPPD Use Peer DNS'));
```
Comment (jow-) [inline protocols/luci-proto-openfortivpn/htdocs/luci-static/resources/protocol/openfortivpn.js:72]: This is not directly related to LuCI but to the underlying proto handler - is there any reason to not use the standard `peerdns` option name which is used by most other proto handlers?

File: protocols/luci-proto-openfortivpn/htdocs/luci-static/resources/protocol/openfortivpn.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	getDevices: function() {
+		return null;
+	},
+
+	containsDevice: function(ifname) {
+		return (network.getIfnameOf(ifname) == this.getIfname());
+	},
+
+	renderFormOptions: function(s) {
+		var dev = this.getDevice().getName(),
```
Comment (jow-) [inline protocols/luci-proto-openfortivpn/htdocs/luci-static/resources/protocol/openfortivpn.js:40]: Remove the `dev = ...` assignment, it appears to be unused.

File: protocols/luci-proto-openfortivpn/htdocs/luci-static/resources/protocol/openfortivpn.js [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+		s.taboption("general", form.Value, "username", _("Username"));
+
+		o = s.taboption('general', form.Value, 'password', _('Password'));
+		o.password = true;
+
+		o = s.taboption('advanced', widgets.NetworkSelect, 'iface_name', _('Bind interface'), _('Bind the tunnel to this interface (optional).'));
+		o.nocreate = true;
+		o.optional = true;
+
+		s.taboption('advanced', form.Value, 'trusted_cert', _("VPN Server's certificate SHA1 hash"));
```
Comment (jow-) [inline protocols/luci-proto-openfortivpn/htdocs/luci-static/resources/protocol/openfortivpn.js:62]: Can we somehow tighten the validation here? What kind of input is required? A 40 byte hex string? If so, try using `o.datatype = 'and(hexstring,length(40))';`

File: protocols/luci-proto-openfortivpn/htdocs/luci-static/resources/protocol/openfortivpn.js (hunk 2 above)
Comment (aaronjg) [reply protocols/luci-proto-openfortivpn/htdocs/luci-static/resources/protocol/openfortivpn.js:72]: good point. I was mirroring the option names for the underlying openfortivpn program. I can make a PR to update the package to use the openwrt standard option.

**Comment by aaronjg** [comment]
I updated the PR with your suggestions - this will also now require an update the the openfortivpn package to use the 'peerdns' configuration option. (PR12799)

**Comment by aaronjg** [comment]
@jow- How does it look now? peerdns commit has been merged into master on openwrt/packages, so I think this is ready to land if it looks okay to you

**Comment by jow-** [comment]
@aaronjg - looks almost perfect, with a little typo in `defaultroute` option (Use default*t* gateway)

**Comment by aaronjg** [comment]
@jow-  Thank you for catching that. Fixed now.

**Comment by feckert** [comment]
From my point of view all this should fit now.
@aaronjg thanks
